internal/scaffold/helm/watches.go

```diff
@@ -43,6 +54,44 @@ func (s *WatchesYAML) GetInput() (input.Input, error) {
 	return s.Input, nil
 }
 
+func UpdateHelmWatchForResource(r *scaffold.Resource, absProjectPath string) error {
+
+	watchFilePath := filepath.Join(absProjectPath, WatchesYamlFile)
+	watchYAML, err := ioutil.ReadFile(watchFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read watch manifest %v: %v", watchFilePath, err)
+	}
+	watchList := []watches.Watch{}
+	err = yaml.Unmarshal(watchYAML, &watchList)
+	if err != nil {
+		log.Error(err, "Failed to unmarshal watch config")
+		return nil
+	}
+	gvk := schema.GroupVersionKind{
+		Version: r.Version,
+		Group:   r.FullGroup,
+		Kind:    r.Kind,
+	}
+	for _, watch := range watchList {
+		if watch.GroupVersionKind == gvk {
+			// dupe detected
+			dupeWatch = true
+			return fmt.Errorf("duplicate GVK: %v", watch.GroupVersionKind.String())
+		}
+	}
+	if dupeWatch == false {
+		data := []byte(string(watchYAML) + "\n\n" +
+			"- version: " + r.Version + "\n" +
+			"  group: " + r.FullGroup + "\n" +
+			"  kind: " + r.Kind + "\n" +
+			"  chart: " + HelmChartsDir + "/" + r.LowerKind)
+		if err := ioutil.WriteFile(watchFilePath, []byte(data), fileutil.DefaultFileMode); err != nil {
```

Comment:
Could we not here persist the json instead of manipulating strings?
Then, it means that the watch file would be a struct with a map of watch objects. 
So, we will load the config and add a new item in the list and then save the changes. WDYT?

Reply:
Agree.

Reply:
@camilamacedo86 structs have been added to add new Watch, in both ansible and helm implementations.